src/machine/flash.go

```diff
@@ -66,21 +66,25 @@ type BlockDevice interface {
 
 // FlashBuffer implements the ReadWriteCloser interface using the BlockDevice interface.
 type FlashBuffer struct {
-	b       BlockDevice
-	start   uintptr
-	current uintptr
+	b BlockDevice
+
+	// start is actual address
+	start uintptr
+
+	// offset is relative to start
+	offset uintptr
 }
 
 // OpenFlashBuffer opens a FlashBuffer.
 func OpenFlashBuffer(b BlockDevice, address uintptr) *FlashBuffer {
-	return &FlashBuffer{b: b, start: address, current: address}
+	return &FlashBuffer{b: b, start: address}
 }
 
 // Read data from a FlashBuffer.
 func (fl *FlashBuffer) Read(p []byte) (n int, err error) {
-	fl.b.ReadAt(p, int64(fl.current))
+	fl.b.ReadAt(p, int64(fl.offset))
```

Comment:
This just swallows errors, should check for error and reflect it out?

Reply:
Good point. Done.